tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -984,6 +998,275 @@ TEST_P(DistributedTransformerTest, Forward) {
+  TensorView* mha_b0 = makeContigConcreteTensor({D, 3 * E / D}, dtype);
+  TensorView* mha_w1 = makeContigConcreteTensor({D, E / D, E}, dtype);
+  TensorView* mha_b1 = makeContigConcreteTensor({E}, dtype);
+  TensorView* mlp_w0 = makeContigTensor(3, dtype);
+  TensorView* mlp_b0 = makeContigTensor(2, dtype);
+  TensorView* mlp_w1 = makeContigTensor(3, dtype);
+  TensorView* mlp_b1 = makeContigTensor(1, dtype);
+  TensorView* mha_mask = makeContigTensor(2, DataType::Bool);
+  TensorView* mlp_mask = makeContigTensor(2, DataType::Bool);
+  TensorView* mha_sdpa_out = makeConcreteTensor({D, B, H / D, S, E / H}, dtype);
+  TensorView* mha_sdpa_log_sumexp =
+      makeContigConcreteTensor({D, B, H / D, S}, DataType::Float);
+  TensorView* mha_sdpa_seed = makeSymbolicTensor({}, DataType::Int);
+  TensorView* mha_sdpa_offset = makeSymbolicTensor({}, DataType::Int);
+  TensorView* ln1_w = makeContigTensor(1);
+  TensorView* ln1_b = makeContigTensor(1);
+  TensorView* ln1_mean = makeConcreteTensor({256, 1});
+  TensorView* ln1_rstd = makeConcreteTensor({256, 1});
+  TensorView* ln0_w = makeContigTensor(1);
+  TensorView* ln0_b = makeContigTensor(1);
+  TensorView* ln0_mean = makeConcreteTensor({256, 1});
+  TensorView* ln0_rstd = makeConcreteTensor({256, 1});
+  TensorView* mha_linear1 = makeContigTensor(2);
+
+  fusion->addInput(x);
+  fusion->addInput(grad);
+  fusion->addInput(mha_w0);
+  fusion->addInput(mha_b0);
+  fusion->addInput(mha_w1);
+  fusion->addInput(mha_b1);
+  fusion->addInput(mlp_w0);
+  fusion->addInput(mlp_b0);
+  fusion->addInput(mlp_w1);
+  fusion->addInput(mlp_b1);
+  fusion->addInput(mlp_mask);
+  fusion->addInput(mha_mask);
+  fusion->addInput(mha_sdpa_out);
+  fusion->addInput(mha_sdpa_log_sumexp);
+  fusion->addInput(mha_sdpa_seed);
+  fusion->addInput(mha_sdpa_offset);
+  fusion->addInput(ln1_w);
+  fusion->addInput(ln1_b);
+  fusion->addInput(ln1_mean);
+  fusion->addInput(ln1_rstd);
+  fusion->addInput(ln0_w);
+  fusion->addInput(ln0_b);
+  fusion->addInput(ln0_mean);
+  fusion->addInput(ln0_rstd);
+  fusion->addInput(mha_linear1);
+
+  const auto D = mha_w0->axis(0)->extent()->value().as<int64_t>();
+  // Recomputation
+  auto ln_0 = layer_norm(x, norm_shape, ln0_w, ln0_b, eps);
+  auto mha_in = castOp(dtype, std::get<0>(ln_0));
+  auto qkv = mha_qkv(mha_in, mha_w0, mha_b0, mha_w1, mesh);
+  // The thunder trace recompute mha linear1, but this would result in 3
+  // AllReduces in the backwards pass.
+  if (mha_linear1 == nullptr) {
+    TensorView* sdpa_transpose = transpose(mha_sdpa_out, 2, 3);
+    TensorView* sdpa_reshape =
+        reshape(sdpa_transpose, {D, B, S, H / D, E / H}, {D, B * S, E / D});
+    TensorView* mha_local_matmul1 = matmul(sdpa_reshape, mha_w1);
+    TensorView* mha_matmul1 = sum(mha_local_matmul1, {0}); // allreduce
+    mha_linear1 = add(mha_matmul1, broadcast(mha_b1, {true, false}));
+  }
+  Val* scale = IrBuilder::create<Val>(1.0 / (1.0 - kDropoutProb));
```

Comment:
```suggestion
  Val* dropout_scale = IrBuilder::create<Val>(1.0 / (1.0 - kDropoutProb));
```
so the readers immediately know what this scale is for.